examples/nine-pt-stencil-solver/nine-pt-stencil-solver.cpp

```diff
@@ -0,0 +1,298 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2019, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+/******************************<DESCRIPTION>*********************************
+
+*******************************<DESCRIPTION>*********************************/
+
+#include <array>
+#include <chrono>
+#include <ginkgo/ginkgo.hpp>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+// Stencil values. Ordering can be seen in the main function
+double alpha_c = 10.0 / 3.0;
+double beta_c = -2.0 / 3.0;
+double gamma_c = -1.0 / 6.0;
+
+std::array<double, 9> coefs;
```

Comment:
Global variables should be avoided in C++, especially in this case as they are defined at a later point. For the `alpha_c`, ..., it might actually be fine to have them, but they should be `constexpr`, and only specified as default values:
```C++
constexpr double default_alpha_c = 10.0 / 3.0;
constexpr double default_beta_c = -2.0 / 3.0;
constexpr double default_gamma_c = -1.0 / 6.0;
```

`coefs`, on the other hand, should be passed (by reference) to the functions that need it and the global variable removed.
You even construct it in the `main` method, so there is no need for a global variable (which causes confusion because it is not really clear where the values are set).

Reply:
I didn't fully migrate it when adding the changable stencil values.
Did it now.